synapse/storage/databases/state/store.py

```diff
@@ -400,10 +400,13 @@ async def store_state_group(
         room_id: str,
         prev_group: Optional[int],
         delta_ids: Optional[StateMap[str]],
-        current_state_ids: StateMap[str],
+        current_state_ids: Optional[StateMap[str]],
     ) -> int:
         """Store a new set of state, returning a newly assigned state group.
 
+        If `current_state_ids` is None then `prev_group` and `delta_ids` must
+        not be None.
+
         Args:
             event_id: The event ID for which the state was calculated
             room_id
```

Comment:
below here it says that `prev_group` is optional but doesn't say the same for `current_state_ids`, which is misleading.

I'd remove the "optional" from the description of `prev_group` in favour of the above.